app/src/main/java/org/schabi/newpipe/player/Player.java

```diff
@@ -860,9 +860,15 @@ public void setRecovery() {
 
         final int queuePos = playQueue.getIndex();
         final long windowPos = simpleExoPlayer.getCurrentPosition();
+        final long duration = simpleExoPlayer.getDuration();
 
-        if (windowPos > 0 && windowPos <= simpleExoPlayer.getDuration()) {
-            setRecovery(queuePos, windowPos);
+        if (windowPos > 0
+                // Sometimes (e.g. when the playback ended) the windowPos is a few milliseconds
+                // higher than the duration. Due to this a little buffer (100ms) was introduced.
+                // See also https://github.com/TeamNewPipe/NewPipe/pull/7195#issuecomment-962624380
+                && windowPos <= duration + 100
+        ) {
+            setRecovery(queuePos, Math.min(windowPos, duration));
```

Comment:
Is the check for `windowPos <= duration` even needed? Due to the `Math.min` no value bigger than `duration` can be passed onto `setRecovery`
In theory `windowPos` should only go from 0 to `duration`. Since it apparently does not do it, is there a difference in meaning from a `windowPos` that is  100 over `duration` and one that is 200 over `duration`

```suggestion
        if (windowPos > 0) {
            // Sometimes (e.g. when the playback ended) the windowPos is a few milliseconds
            // higher than the duration.
            // See also https://github.com/TeamNewPipe/NewPipe/pull/7195#issuecomment-962624380
            setRecovery(queuePos, Math.min(windowPos, duration));
```

Reply:
> Is the check for windowPos <= duration even needed?

I didn't implement it and left no comment and context (git blame also didn't really help since the code was copied more than 2 times) - like the author before - so I assume it does something useful.

However if we do this we can also do ``setRecovery(queuePos, Math.max(0, Math.min(windowPos, duration)));`` so that there are no checks at all 😄

Reply:
> However if we do this we can also do setRecovery(queuePos, Math.max(0, Math.min(windowPos, duration))); so that there are no checks at all 😄

This seems to also fix the bug. But if that breaks something else is something is another question which I can't answer.

Reply:
I removed the check completely.
Everything seems to be working as expected.